src/Nethermind/Nethermind.JsonRpc/WebSockets/JsonRpcSocketsClient.cs

```diff
@@ -111,42 +107,64 @@ private void IncrementBytesSentMetric(int size)
         }
 
         public virtual async Task<int> SendJsonRpcResult(JsonRpcResult result)
+        {
+            if (result.IsCollection)
+            {
+                int singleResponseSize = 0;
+                bool isFirst = true;
+                await _handler.SendRawAsync(_jsonOpeningBracket);
+                await foreach (JsonRpcResult.Entry innerResult in result.BatchedResponses!)
+                {
+                    if (!isFirst) await _handler.SendRawAsync(_jsonComma);
+                    isFirst = false;
+
+                    singleResponseSize += await SendJsonRpcResultEntry(innerResult);
+                    _jsonRpcLocalStats.ReportCall(innerResult.Report);
+                }
+                await _handler.SendRawAsync(_jsonClosingBracket);
+
+                return singleResponseSize;
+            }
+            else
+            {
+                int singleResponseSize = await SendJsonRpcResultEntry(result.Response!.Value);
+                return singleResponseSize;
+            }
+        }
+
+        private async Task<int> SendJsonRpcResultEntry(JsonRpcResult.Entry result)
         {
             void SerializeTimeoutException(MemoryStream stream)
             {
                 JsonRpcErrorResponse error = _jsonRpcService.GetErrorResponse(ErrorCodes.Timeout, "Request was canceled due to enabled timeout.");
                 _jsonSerializer.Serialize(stream, error);
             }
 
-            await using MemoryStream resultData = new();
-
-            try
+            using (result)
             {
-                if (result.IsCollection)
+                await using MemoryStream resultData = new();
```

Comment:
As we are doing such a great job already, maybe we can drop this allocation too? To some reused buffer.

Reply:
I guess we can.

Reply:
We can do in next PR though

Reply:
My ideas are to implement either `Stream` or `TextWriter` based on `IByteBuffer`